lib/gtfs/components/GtfsMap.js

```diff
@@ -213,15 +219,23 @@ export default class GtfsMap extends Component<Props, State> {
   _onMarkerDragEnd = (e: any) => { this.fetchIsochrones(e.target.getLatLng()) }
 
   layerAddHandler = (e: any) => {
+    const {stop, pattern} = this.props
+    const {searchFocus} = this.state
     // handle pattern panTo and popup open
-    const patternId = e.layer.feature && e.layer.feature.properties.patternId
-    if (patternId && patternId === this.props.searchFocus) {
-      this.refs.map && this.refs.map.leafletElement.fitBounds(e.layer.getBounds())
-      e.layer.openPopup && e.layer.openPopup()
+    if (pattern && pattern.pattern_id === searchFocus) {
+      // Wait to call open pop up to ensure that layer has fully materialized
+      window.setTimeout(() => {
+        if (e.layer.getBounds && this.refs.map) {
+          // Fit bounds to polyline if function exists
+          this.refs.map.leafletElement.fitBounds(e.layer.getBounds())
+        }
+        // Open popup for pattern if added via a selection from the search
+        e.layer.openPopup && e.layer.openPopup()
```

Comment:
Is there a way to do this stuff purely with react-leaflet?

Reply:
I'm not sure.